homeassistant/components/switch/homekit_controller.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for Homekit switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.homekit/
+"""
+import json
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+
+REQUIREMENTS = ['homekit==0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, service):
+    """Set up Homekit lighting."""
+    add_devices([HomekitSwitch(service)])
+
+
+class HomekitSwitch(SwitchDevice):
+    """Representation of a Homekit switch."""
+
+    def __init__(self, service):
+        """Initialize the switch."""
+
+        self._name = service['device'].model
+        self._securecon = service['device'].securecon
+        self._aid = service['aid']
+        self._iid = service['service']['iid']
+        self._address = "%s-%s" % (service['device'].hkid, service['aid'])
+        self._features = 0
+        self._chars = {}
+        self.update_characteristics(service['service']['characteristics'])
+
+    def update_state(self):
```

Comment:
Same comments as for light:
 - rename to `update`
 - Can we cache the call to `/accessories` between all devices?
 - Call `schedule_update_ha_state` once

Reply:
Largely done